### ISSUE TITLE
[java] AvoidRethrowingException should allow rethrowing exception subclasses

### ISSUE THREAD

**Issue by cowwoc** [pmd/pmd#4911]
**Affects PMD Version:** 7.0.0

**Rule:** https://docs.pmd-code.org/latest/pmd_rules_java_design.html#avoidrethrowingexception

**Description:**
`AvoidRethrowingException` is meant to catch unnecessary catch blocks that simply rethrow the exception. However, there is an edge case that is not considered by the rule.

**Code Sample demonstrating the issue:**

```java
try
{
  Files.delete("example.txt");
}
catch (NoSuchFileException e)
{
  throw e;
}
catch (IOException e)
{
  // An error occurred. Do something special.
}
```

**Expected outcome:**
In the above example, one must catch and rethrow `NoSuchFileException` in order to avoid being caught by the `IOException` handler. Sometimes it is possible to refactor the code to avoid this, but sometimes it is not.

Please update the rule to ignore catching and rethrowing an exception if a later `catch` matches the exception's superclass.

**Running PMD through:** Maven
